dev/src/write-batch.ts

```diff
@@ -567,25 +614,55 @@ export class WriteBatch {
     );
 
     if (explicitTransaction) {
+      assert(!useBatch, 'Cannot use batch mode with transactions');
       request.transaction = explicitTransaction;
     }
 
-    return this._firestore
-      .request<api.ICommitRequest, api.CommitResponse>('commit', request, tag)
-      .then(resp => {
-        if (request.writes!.length > 0) {
-          const commitTime = Timestamp.fromProto(resp.commitTime!);
-          return resp.writeResults.map(
-            writeResult =>
-              new WriteResult(
-                writeResult.updateTime
-                  ? Timestamp.fromProto(writeResult.updateTime)
-                  : commitTime
-              )
-          );
-        }
-        return [];
-      });
+    if (useBatch) {
+      return this._firestore
+        .request<api.IBatchWriteRequest, api.BatchWriteResponse>(
+          'batchWrite',
+          request as api.IBatchWriteRequest,
+          tag
+        )
+        .then(resp => {
+          const zipped: Array<[
```

Comment:
I don't fully understand the need for the intermediate conversion.

Reply:
forgot to refactor 🥇